File: Go/array.go
package main

import (
	"fmt"
	"math/rand"
	"sort"
)

func main() {
	// Create an array of strings.
	randomString := []string{"apple", "jesus", "map", "china", "ass", "bees", "", ""}
	// Check if the array contains the value.
	if arrayContains("apple", randomString) {
		fmt.Println(true)
	}
	//
	for number, list := range randomString {
		// Number
		if number == 25 {
			fmt.Println("The content of number 25 is", list)
		}
		// Item
		if list == "P" {
			fmt.Println("The letter is", list)
		}
	}
	fmt.Println(randomString)
	// Get the index value of a item in a array
	fmt.Println(indexValueInArray("jesus", randomString))
	// Remove all the empty elements in a array.
	fmt.Println(removeEmptyElements(randomString))
	// Check if the array is empty
	fmt.Println(arrayIsEmpty(randomString))
	// Check if the array isnt empty
	fmt.Println(arrayIsntEmpty(randomString))
	// Remove all the duplicate elements in a array.
	fmt.Println(removeDuplicateElements(randomString))
	// Sort the array
	fmt.Println(sortList(randomString))
	// Combine two arrays together.
	fmt.Println(combineArrays(randomString, randomString))
	// Get the median value of an array.
	fmt.Println(medianOfArray([]float64{1, 2, 3, 4, 5, 6, 7, 8, 9, 10}))
	// Reverse an array
	fmt.Println(reverseArray(randomString))
	// Suffle an array
	fmt.Println(shuffleArray(randomString))
	// Get the first value in the array
	fmt.Println(firstValueInArray(randomString))
	// Get the last value in the array
	fmt.Println(lastValueInArray(randomString))
}

// Check if the array contains the value.
func arrayContains(cointains string, originalArray []string) bool {
	for _, value := range originalArray {
		if value == cointains {
			return true
		}
	}
	return false
}

// Remove all the duplicate elements in a array.
func removeDuplicateElements(array []string) []string {
	var newArray []string
	for _, value := range array {
		if !arrayContains(value, newArray) {
			newArray = append(newArray, value)
		}
	}
	return newArray
}

// Get the index value of a item in a array
func indexValueInArray(cointains string, originalArray []string) int {
	for indexValue, value := range originalArray {
		if value == cointains {
			return indexValue
		}
	}
	return 0
}

// Remove all the empty elements in a array.
func removeEmptyElements(array []string) []string {
	var newArray []string
	for _, value := range array {
		if len(value) > 0 {
			newArray = append(newArray, value)
		}
	}
	return newArray
}

// Check if the array is empty
func arrayIsEmpty(array []string) bool {
	if len(array) == 0 {
		return true
	}
	return false
}

// Check if the array isnt empty
func arrayIsntEmpty(array []string) bool {
	if len(array) == 0 {
		return false
	}
	return true
}

// Sort the array
func sortList(content []string) []string {
	sort.Strings(content)
	return content
}

// Combine two arrays together.
func combineArrays(firstArray []string, secondArray []string) []string {
	var newArray []string
	for _, value := range firstArray {
		newArray = append(newArray, value)
	}
	for _, value := range secondArray {
		newArray = append(newArray, value)
	}
	return newArray
}

// Get the median value of an array.
func medianOfArray(array []float64) float64 {
	sort.Float64s(array)
	if len(array)%2 == 0 {
		return (array[len(array)/2-1] + array[len(array)/2]) / 2
	}
	return array[len(array)/2]
}

// Reverse an array
func reverseArray(array []string) []string {
	var newArray []string
	for i := len(array) - 1; i >= 0; i-- {
		newArray = append(newArray, array[i])
	}
	return newArray
}

// Suffle an array
func shuffleArray(array []string) []string {
	for i := range array {
		j := rand.Intn(i + 1)
		array[i], array[j] = array[j], array[i]
	}
	return array
}

// Get the first value in the array
func firstValueInArray(array []string) string {
	return array[0]
}

// Get the last value in the array
func lastValueInArray(array []string) string {
	return array[len(array)-1]
}

File: TypeScript/array.ts
function main() {
    // Create a array of int
    var user_generated_array: string[] = [
        "t",
        "b",
        "l",
        "y",
        "k",
        "i",
        "j",
        "c",
        "a",
        "e",
        "n",
        "d",
        "s",
        "m",
        "p",
        "f",
        "z",
        "r",
        "q",
        "w",
        "u",
        "v",
        "x",
        "o",
        "h",
        "g"
    ]
    // Join the array
    console.log(joinArray(user_generated_array, "-"))
    // Sort the array
    console.log(sortArray(user_generated_array))
    // Get the first value
    console.log(getFirstValue(user_generated_array))
    // Get the last value
    console.log(getLastValue(user_generated_array))
    // Get the length of the array
    console.log(getLength(user_generated_array))
    // Randomly pick a value from the array
    console.log(getRandomValue(user_generated_array))
    // Randomize the order of the values in the array
    console.log(randomizeArray(user_generated_array))
    // Remove the first element in the array.
    console.log(removeFirstElement(user_generated_array))
    // Remove the last element in the array
    console.log(removeLastElement(user_generated_array))
    // Remove a value from the array
    console.log(removeValue(user_generated_array, "y"))
    // Reverse the order of the values in the array
    console.log(reverseArray(user_generated_array))
    // Get the index of a value in the array
    console.log(getIndex(user_generated_array, "x"))
    // Remove all duplicates from an array
    console.log(removeDuplicates(user_generated_array))
    // Combine two arrays
    console.log(combineArrays(user_generated_array, ["six", "seven", "eight"]))
    // Check if the array is empty
    console.log(isArrayEmpty(user_generated_array))
    // Check if the array is not empty
    console.log(isNotEmptyArray(user_generated_array))
    // Remove all the content from the array
    console.log(removeAllContent(user_generated_array))
}

main()

// Join the array.
function joinArray(array: string[], separator: string) {
    return array.join(separator)
}

// Sort all the data in the array, and return the sorted array
function sortArray(array: string[]) {
    array.sort()
    return array
}

// Get the first value in the array and return it
function getFirstValue(array: string[]) {
    return array[0]
}

// Get the last value in the array and return it.
function getLastValue(array: string[]) {
    return array[array.length - 1]
}

// Get the length of the array and return it.
function getLength(array: string[]) {
    return array.length
}

// Randomly pick a value from the array and return it.
function getRandomValue(array: string[]) {
    return array[Math.floor(Math.random() * array.length)]
}

// Randomize the order of the values in the array and return the array.
function randomizeArray(array: string[]) {
    return array.sort(function() {
        return 0.5 - Math.random()
    })
}

// Remove the first element in the array and return the array.
function removeFirstElement(array: string[]) {
    array.shift()
    return array
}

// Remove the last element in the array and return the array.
function removeLastElement(array: string[]) {
    array.pop()
    return array
}

// Remove a value from the array and return the array.
function removeValue(array: string[], value: string) {
    var index = array.indexOf(value)
    if (index > -1) {
        array.splice(index, 1)
    }
    return array
}

// Remove all the content from the array and return the array.
function removeAllContent(array: string[]) {
    array.length = 0
    return array
}

// Reverse the order of the values in the array and return the array.
function reverseArray(array: string[]) {
    return array.reverse()
}

// Get the index of a value in the array and return it.
function getIndex(array: string[], value: string) {
    return array.indexOf(value)
}

// Remove all duplicates from an array and return the array.
function removeDuplicates(array: string[]) {
    var newArray: string[] = []
    for (var i = 0; i < array.length; i++) {
        if (newArray.indexOf(array[i]) == -1) {
            newArray.push(array[i])
        }
    }
    return newArray
}

// Combine two arrays and return the array.
function combineArrays(array1: string[], array2: string[]) {
    return array1.concat(array2)
}

// Check if the array is empty.
function isArrayEmpty(array: string[]) {
    return array.length == 0
}

// Check if the array is not empty.
function isNotEmptyArray(array: string[]) {
    return array.length != 0
}
